nrf52-hal-common/src/uicr.rs

```diff
@@ -0,0 +1,56 @@
+use crate::target::{NVMC, UICR};
+
+/// Interface to a UICR instance
+///
+/// This is a very basic interface that comes with the following limitations:
+/// - Only `customer` registers are usable for storing and loading of data
+/// - Erase must be performed in order to write bits with value `1` over `0`
+pub struct Uicr(UICR);
+
+impl Uicr {
+    /// Construct a new `Uicr` from `pac::UICR`
+    pub fn new(uicr: UICR) -> Self {
+        Self(uicr)
+    }
+
+    /// Release the `pac::UICR` instance back
+    pub fn free(self) -> UICR {
+        self.0
+    }
+
+    /// Erase the UICR registers. UICR registers can only be set to `0` bits, additional
+    /// overrides back to `1` can only be performed by erasing the UICR registers.
+    /// - Sets all registers to 0xFFFFu32
+    pub fn erase(&mut self, nvmc: &mut NVMC) {
+        assert!(nvmc.config.read().wen().is_wen() == false); // write + erase is forbidden!
+
+        nvmc.config.write(|w| w.wen().een());
+        nvmc.eraseuicr.write(|w| w.eraseuicr().erase());
+        nvmc.config.reset()
+    }
+
+    /// Store a slice of `&[u32]` values to the customer registers with given offset
+    /// - offset + slice length must be less than 32
```

Comment:
This should probably mention that the new configuration only takes effect after a reset (according to the manual), otherwise that seems like an easy-to-miss footgun

Reply:
Well it's a bit more nuanced than that. You can technically set say `0b01101111` for example and then override to `0b00001111` or such no problems. It's only `1` that cannot be set after a `0`. I can add more strong warnings here tho.

Reply:
Ah, that's not what I meant – the manual says that the *chip* needs to be reset for the written UICR data to "take effect". Or does that only refer to the non-customer data like the reset/NFC pin configuration?

(still sounds like a good idea to mention that an erase might be needed though)

Reply:
Hmm I think it's only for the "functional" registers. I did do a store and load test with my quicktime setup and it worked fine, I got the stored value back.

Reply:
Okay, it would be very weird if even the customer data would require a reset, so let's not mention that in the docs.